### PR TITLE
Holdings error fix

### PR DESCRIPTION
**Pull request by dbwwei**
Fixed a bug in availability_response.rb to handle empty/nill key or content; added respective tests to multiple_bibs.json.

### CODE REVIEW

File: spec/fixtures/multiple_bibs.json [hunk 1]
```diff
@@ -70,7 +70,7 @@
...
...field></datafield><datafield ind1=\" \" ind2=\" \" tag=\"997\"><subfield code=\"a\">20180220</subfield><subfield code=\"d\">TEU</subfield><subfield code=\"c\">COUTTS SR</subfield><subfield code=\"e\">MARCIVE</subfield><subfield code=\"e\">WCAT</subfield></datafield><datafield ind1=\" \" ind2=\" \" tag=\"INT\"><subfield code=\"a\">P</subfield></datafield><datafield ind1=\" \" ind2=\" \" tag=\"INST\"><subfield code=\"a\">01TULI_INST</subfield></datafield><datafield ind1=\" \" ind2=\" \" tag=\"AVA\"><subfield code=\"0\">991036880906903811</subfield><subfield code=\"8\">22413618700003811</subfield><subfield code=\"a\">01TULI_INST</subfield><subfield code=\"b\">MAIN</subfield><subfield code=\"c\">Stacks</subfield><subfield code=\"d\">PN4775 .F45 2019</subfield><subfield code=\"e\">available</subfield><subfield code=\"f\">1</subfield><subfield code=\"g\">0</subfield><subfield code=\"i\">m</subfield><subfield code=\"j\">stacks</subfield><subfield code=\"k\">0</subfield><subfield code=\"p\">1</subfield><subfield code=\"q\">Paley Library</subfield><subfield code=\"t\">holding-1</subfield><subfield code=\"t\">holding-2</subfield><subfield code=\"t\">holding-3</subfield></datafield></record>"
```
Comment (mksndz) [inline spec/fixtures/multiple_bibs.json:73]: Are these `nill` subfield values supposed to evaluate to a `nil` value when parsed?

Comment (mksndz) [reply spec/fixtures/multiple_bibs.json:73]: I haven't run this code, but I suspect another approach is needed to ensure that actual `nil` values are properly handled.

Comment (dbwwei) [reply spec/fixtures/multiple_bibs.json:73]: Basically two questions: if the code is empty, or not defined, shall it be ignored? if the content is empty, shall it be ignored?

Comment (dbwwei) [reply spec/fixtures/multiple_bibs.json:73]: [record-991853173503681.json](https://github.com/user-attachments/files/17805802/record-991853173503681.json)

Comment (dbwwei) [reply spec/fixtures/multiple_bibs.json:73]: This is the record the causes problem: there are four cases of: {"code"=>"t"}. Without the "content", they will cause error.

Comment (dbwwei) [reply spec/fixtures/multiple_bibs.json:73]: To avoid any confusion, I've updated file multiple_bibs.json to add only one case of <subfield code="t"></subfield> to test for empty content. The other subfields I added to the file last time are all removed.

**Comment by cgalarza** [comment]
@dbwwei We need more explicit tests in this PR that cover the cases this change is trying to fix. The availability spec has some testing but doesn't have explicit tests for checking when a value is not present and/or blank.

File: spec/lib/availability_response_spec.rb [hunk 2]
```diff
@@ -19,60 +19,68 @@
...
-        "total_items" => "1",
-        "non_available_items" => "0",
-        "location_code" => "stacks",
-        "call_number_type" => "0",
-        "priority" => "1",
-        "library" => "Paley Library",
-        "inventory_type" => "physical",
-        "holding_info" => "holding-1 holding-2 holding-3" }]
+    it "skips the empty holding info (code=t) subfield" do
+      expect(availability["991023558879703811"][:holdings][0]["holding_info"]).to be_nil
```
Comment (tampakis) [inline spec/lib/availability_response_spec.rb]: @dbwwei The format of the fixture for `991023558879703811` doesn't match what you are testing for. It doesn't have a subfield with code=t. I like that we have this test as a base case for records that don't have holding info included.

There should be an additional fixture that includes a subfield code=t but has no content inside it.

Other scenarios could be a record with a single subfield code=t with content in it and a record with multiple subfields code=t where one of them is missing content.

It should be possible to add more fixture records in the [multiple_bibs.json](https://github.com/tulibraries/alma_rb/blob/b291f2489ec674dbde8475a0da4c45a3896072c0/spec/fixtures/multiple_bibs.json) to capture these additional scenarios. Make sure they have distinct IDs. Also since MARCXML string is hard to read in the "anies" field, could you try formatting the string to span across multiple lines in the file (does it work if you add `\n` at the ends of lines?) and only including the essential bits of the record (removing all of the datafields except the 001 and the AVA tag) so it's easier to tell what is actually in the fixture data?

Comment (dbwwei) [reply spec/lib/availability_response_spec.rb]: I've added five new test cases with the minimal for testing.

**Review by sensei100 — APPROVED** [review]
